--- qt.py	2013-09-09 23:36:34.000000000 +0200
+++ qt.py.patched	2013-11-08 11:32:59.164479101 +0100
@@ -9,14 +9,14 @@
 import os
 
 from IPython.external.qt_loaders import (load_qt, QT_API_PYSIDE,
-                                         QT_API_PYQT)
+                                         QT_API_PYQT, loaded_api)
 
 QT_API = os.environ.get('QT_API', None)
 if QT_API not in [QT_API_PYSIDE, QT_API_PYQT, None]:
     raise RuntimeError("Invalid Qt API %r, valid values are: %r, %r" %
                        (QT_API, QT_API_PYSIDE, QT_API_PYQT))
 if QT_API is None:
-    api_opts = [QT_API_PYSIDE, QT_API_PYQT]
+    api_opts = [QT_API_PYSIDE, QT_API_PYQT, loaded_api()]
 else:
     api_opts = [QT_API]
 
