--- eulxml-1.1.3/eulxml/xpath/core.py	2017-01-25 09:59:18.751452624 +0100
+++ eulxml-1.1.3/eulxml/xpath/core.py	2016-02-19 23:15:24.000000000 +0100
@@ -143,7 +143,7 @@
 # If we don't have write permission, put them in the configured tempdir
 if (not os.access(parsedir, os.W_OK)):
     parsedir = tempfile.gettempdir()
-parser = yacc.yacc(module=parserules, outputdir=parsedir, debug=0)
+parser = yacc.yacc(module=parserules, outputdir=parsedir, debug=0, optimize=1)
 
 
 def parse(xpath):
